Document/0x05d-Testing-Data-Storage.md

```diff
@@ -509,21 +421,181 @@ There are several different open-source libraries that offer encryption capabili
+  val xorParts0 = Base64.decode(myCompositeKey[0], 0)
+  val xorParts1 = Base64.decode(myCompositeKey[1], 0)
+  val xorKey = ByteArray(xorParts0.size)
+  for (i in xorParts1.indices)
+  {
+    xorKey[i] = (xorParts0[i] xor xorParts1[i]).toByte()
+  }
+  HidingUtil.doHiding(myHiddenMessage.toByteArray(), xorKey, false)
+}
+```
+
+Verify common locations of secrets:
+
+- resources (typically at res/values/strings.xml)
+  Example:
+
+  ```xml
+  <resources>
+      <string name="app_name">SuperApp</string>
+      <string name="hello_world">Hello world!</string>
+      <string name="action_settings">Settings</string>
+      <string name="secret_key">My_Secret_Key</string>
+    </resources>
+  ```
+
+- build configs, such as in local.properties or gradle.properties
+  Example:
+
+  ```default
+  buildTypes {
+    debug {
+      minifyEnabled true
+      buildConfigField "String", "hiddenPassword", "\"${hiddenPassword}\""
+    }
+  }
+  ```
+
 ### Dynamic Analysis
 
 Install and use the app, executing all functions at least once. Data can be generated when entered by the user, sent by the endpoint, or shipped with the app. Then complete the following:
 
+- Check both internal and external local storage for any files created by the application that contain sensitive data.
 - Identify development files, backup files, and old files that shouldn't be included with a production release.
 - Determine whether SQLite databases are available and whether they contain sensitive information. SQLite databases are stored in `/data/data/<package-name>/databases`.
-- Check Shared Preferences that are stored as XML files (in `/data/data/<package-name>/shared_prefs`) for sensitive information. Avoid using Shared Preferences and other mechanisms that can't protect data when you are storing sensitive information. Shared Preferences is insecure and unencrypted by default. You can use [secure-preferences](https://github.com/scottyab/secure-preferences "Secure-preferences encrypts the values of Shared Preferences") to encrypt the values stored in Shared Preferences, but the Android KeyStore should be your first choice for storing data securely.
+- Identify if SQLite databases are encrypted. If so, determine how the database password is generated and stored and if this is sufficiently protected as described in the "[Storing a Key](#storing-a-key)" section of the Keystore overview.
+- Check Shared Preferences that are stored as XML files (in `/data/data/<package-name>/shared_prefs`) for sensitive information. Avoid using Shared Preferences and other mechanisms that can't protect data when you are storing sensitive information. Shared Preferences is insecure and unencrypted by default. You can use [secure-preferences](https://github.com/scottyab/secure-preferences "Secure-preferences encrypts the values of Shared Preferences") to encrypt the values stored in Shared Preferences, but the Android KeyStore should be your first choice for storing data locally securely.
```

Comment:
some part removed since it appears below as a recommendation, this should only focus on what to check.
```suggestion
- Check Shared Preferences that are stored as XML files (in `/data/data/<package-name>/shared_prefs`) for sensitive information. Shared Preferences are insecure and unencrypted by default. Some apps might opt to use [secure-preferences](https://github.com/scottyab/secure-preferences "Secure-preferences encrypts the values of Shared Preferences") to encrypt the values stored in Shared Preferences.
```